python_on_whales/components/pod/cli_wrapper.py

```diff
@@ -0,0 +1,778 @@
+        return json.loads(json_str)
+
+    def _parse_json_object(self, json_object: Mapping[str, Any]) -> PodInspectResult:
+        return PodInspectResult(**json_object)
+
+    def _get_inspect_result(self) -> PodInspectResult:
+        """Only there to allow tools to know the return type"""
+        return super()._get_inspect_result()
+
+    @property
+    def id(self) -> str:
+        return self._get_immutable_id()
+
+    @property
+    def name(self) -> str:
+        return self._get_inspect_result().name
+
+    @property
+    def created(self) -> datetime:
+        return self._get_inspect_result().created
+
+    @property
+    def create_command(self) -> List[str]:
+        return self._get_inspect_result().create_command
+
+    @property
+    def exit_policy(self) -> str:
+        return self._get_inspect_result().exit_policy
+
+    @property
+    def state(self) -> str:
+        return self._get_inspect_result().state
+
+    @property
+    def hostname(self) -> str:
+        return self._get_inspect_result().hostname
+
+    @property
+    def create_cgroup(self) -> bool:
+        return self._get_inspect_result().create_cgroup
+
+    @property
+    def cgroup_parent(self) -> str:
+        return self._get_inspect_result().cgroup_parent
+
+    @property
+    def cgroup_path(self) -> str:
+        return self._get_inspect_result().cgroup_path
+
+    @property
+    def create_infra(self) -> bool:
+        return self._get_inspect_result().create_infra
+
+    @property
+    def infra_container_id(self) -> str:
+        return self._get_inspect_result().infra_container_id
+
+    @property
+    def infra_config(self) -> PodInfraConfig:
+        return self._get_inspect_result().infra_config
+
+    @property
+    def shared_namespaces(self) -> List[str]:
+        return self._get_inspect_result().shared_namespaces
+
+    @property
+    def num_containers(self) -> int:
+        return self._get_inspect_result().num_containers
+
+    @property
+    def containers(self) -> List[PodContainer]:
+        return self._get_inspect_result().containers
+
+    def __repr__(self):
+        return f"python_on_whales.Pod(id='{self.id[:20]}', name={self.name})"
+
+    def exists(self) -> bool:
+        """Returns `True` if the pod exists and `False` if not.
+
+        See the `docker.pod.exists` command for information about the arguments.
+        """
+        return PodCLI(self.client_config).exists(self.id)
+
+    def kill(self, *, signal: Optional[Union[int, str]] = None) -> None:
+        """Kill this pod
+
+        See the [`docker.pod.kill`](../sub-commands/container.md#kill) command for
+        information about the arguments.
+        """
+        return PodCLI(self.client_config).kill(self, signal=signal)
+
+    def logs(
+        self,
+        *,
+        container: Optional[
+            python_on_whales.components.container.cli_wrapper.ValidContainer
+        ] = None,
+        names: bool = False,
```

Comment:
```suggestion
        container: Optional[
            python_on_whales.components.container.cli_wrapper.ValidContainer
        ] = None,
        *,
        names: bool = False,
```

I think we should allow `my_pod.logs(my_container_name)`, it doesn't necessitate additional information to be understood.